### PR TITLE
reintroduce binary compatabilty lost in #303

### PR DESCRIPTION
**Pull request by jtnord**
as yet untest approach to reintroce binary api compatability lost in #303 without the need to update several other plugins.

- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by jtnord** [comment]
@jglick trying to test this manually - but in the original you say 
> Otherwise clicking the sidepanel entries for term / kill sends you to a blank page.

However I do not see any side panel entries for that - so unsure how you are reaching it.

image

image


there is `doStop` from the red x, but I think you are referring to something else?

**Comment by jglick** [comment]
Yes I am referring to a black **X** icon in the sidepanel. https://github.com/jenkinsci/workflow-job-plugin/blob/9b3f35756381efcc95550084320e342e903c8609/src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowRun/sidepanel.jelly#L55-L74 I seem to recall there being some conditions under which these were not shown.

File: src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java [hunk 1]
```diff
@@ -872,7 +889,8 @@ public synchronized HttpResponse doStop() {
         if (e != null) {
             return e.doStop();
         } else {
-            return doKill();
+            doKill();
+            return HttpResponses.forwardToPreviousPage();
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java:893]: Does the double response break anything?

**Review by jglick — APPROVED** [review]
_Looks_ like it should work.

**Comment by jglick** [comment]
Another approach I thought of was e.g.

```java
public void doTerm() {
  // as previously
}
@RequirePOST
@WebMethod("term") public HttpResponse httpTerm() {
  doTerm();
  return HttpResponses.forwardToPreviousPage();
}
```

if Stapler in fact likes that.

File: src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java (hunk 1 above)
Comment (jtnord) [reply src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java:893]: 🤔  possibly a stack trace that a response has been committed.  I need to manually confirm :)

**Comment by jtnord** [comment]
tested with the following pipleine

```
while (true) {
    try {
        sleep 500000
    } catch(e) {}
}
```
image


this caused the links to show up in the sidepanel.  they worked without any stack showing in the output of `mvn hpi:run`

I can investigate the alternative if you wish?

**Pull request closed without merging**